arch/cpu/cc2538/dev/spi-arch.c

```diff
@@ -0,0 +1,373 @@
+                          GPIO_PIN_MASK(mosi_pin));
+  GPIO_PERIPHERAL_CONTROL(GPIO_PORT_TO_BASE(miso_port),
+                          GPIO_PIN_MASK(miso_pin));
+
+  /* Disable any pull ups or the like */
+  ioc_set_over(clk_port, clk_pin, IOC_OVERRIDE_DIS);
+  ioc_set_over(mosi_port, mosi_pin, IOC_OVERRIDE_DIS);
+  ioc_set_over(miso_port, miso_pin, IOC_OVERRIDE_DIS);
+
+  /* Configure the clock */
+  REG(regs->base + SSI_CPSR) = regs->ssi_cprs_cpsdvsr;
+
+  /* Configure the mode */
+  REG(regs->base + SSI_CR0) = mode | (0x07);
+
+  /* Configure the SSI serial clock rate */
+  if(!dev->spi_bit_rate) {
+    scr = 255;
+  } else {
+    div = (uint64_t)regs->ssi_cprs_cpsdvsr * dev->spi_bit_rate;
+    scr = (SSI_SYS_CLOCK + div - 1) / div;
+    scr = MIN(MAX(scr, 1), 256) - 1;
+  }
+  REG(regs->base + SSI_CR0) = (REG(regs->base + SSI_CR0) & ~SSI_CR0_SCR_M) |
+    scr << SSI_CR0_SCR_S;
+
+  /* Enable the SSI */
+  REG(regs->base + SSI_CR1) |= SSI_CR1_SSE;
+
+  return SPI_DEV_STATUS_OK;
+}
+/*---------------------------------------------------------------------------*/
+spi_status_t
+spi_arch_close(spi_device_t *dev)
+{
+  if(!spi_arch_has_lock(dev)) {
+    return SPI_DEV_STATUS_BUS_NOT_OWNED;
+  }
+
+  /* Disable SSI */
+  REG(SYS_CTRL_RCGCSSI) &= ~(1 << dev->spi_controller);
+
+  return SPI_DEV_STATUS_OK;
+}
+/*---------------------------------------------------------------------------*/
+spi_status_t
+spi_arch_select(spi_device_t *dev)
+{
+
+  if(!spi_arch_has_lock(dev)) {
+    return SPI_DEV_STATUS_BUS_NOT_OWNED;
+  }
+
+  SPIX_CS_CLR(PIN_TO_PORT(dev->pin_spi_cs), PIN_TO_NUM(dev->pin_spi_cs));
+
+  return SPI_DEV_STATUS_OK;
+}
+/*---------------------------------------------------------------------------*/
+spi_status_t
+spi_arch_deselect(spi_device_t *dev)
+{
+  SPIX_CS_SET(PIN_TO_PORT(dev->pin_spi_cs), PIN_TO_NUM(dev->pin_spi_cs));
+
+  return SPI_DEV_STATUS_OK;
+}
+/*---------------------------------------------------------------------------*/
+/* Assumes that checking dev and bus is not NULL before calling this */
+spi_status_t
+spi_arch_transfer(spi_device_t *dev,
+                  const uint8_t *write_buf, int wlen,
+                  uint8_t *inbuf, int rlen, int ignore_len)
+{
+  int i;
+  int totlen;
+  uint8_t c;
+
+  if(!spi_arch_has_lock(dev)) {
+    return SPI_DEV_STATUS_BUS_NOT_OWNED;
+  }
+
+  LOG_DBG("SPI: transfer (r:%d,w:%d) ", rlen, wlen);
+
+  if(write_buf == NULL && wlen > 0) {
+    return SPI_DEV_STATUS_EINVAL;
+  }
+  if(inbuf == NULL && rlen > 0) {
+    return SPI_DEV_STATUS_EINVAL;
+  }
+
+  totlen = MAX(rlen + ignore_len, wlen);
+
+  if(totlen == 0) {
+    /* Nothing to do */
+    return SPI_DEV_STATUS_OK;
+  }
+
+  LOG_DBG("%c%c%c: %u ", rlen > 0 ? 'R' : '-', wlen > 0 ? 'W' : '-',
+          ignore_len > 0 ? 'S' : '-', totlen);
+
+  for(i = 0; i < totlen; i++) {
+    spix_wait_tx_ready(dev);
+    c = i < wlen ? write_buf[i] : 0;
+    spix_write_buf(dev, c);
+    LOG_DBG("%c%02x->", i < rlen ? ' ' : '#', c);
```

Comment:
This should be `LOG_DBG_()` instead of `LOG_DBG()` to avoid a new log line with prefix. Same for other debug output in this function.

Reply:
I'll fix this thanks.